src/charm.py

```diff
@@ -66,6 +65,14 @@ def __init__(self, *args: typing.Any) -> None:
         )
         self._observability = Observability(charm=self, charm_state=self._charm_state)
 
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(
+            self.on.statsd_prometheus_exporter_pebble_ready,
+            self._on_statsd_prometheus_exporter_pebble_ready,
+        )
+        self.framework.observe(self.on.rotate_secret_key_action, self._on_rotate_secret_key_action)
+        self.framework.observe(self.on.secret_storage_relation_changed, self._on_config_changed)
```

Comment:
Each event should have its own handler. They can delegate to shared methods for code re-usability

Reply:
Updated, thanks!